modules/src/clients/ics07_tendermint/client_def.rs

```diff
@@ -43,15 +43,21 @@ pub struct TendermintClient {
 
 impl ClientDef for TendermintClient {
     type ClientState = TmClientState;
-    type ConsensusState = TmConsensusState;
+
+    fn validate_consensus_state(
```

Comment:
I don't see this as validation; it's more "delayed deserialization" to me. How about simply `box_consensus_state()`?

Reply:
I named it `validate_consensus_state()` because other light clients may choose to perform additional validation/initialization here that is specific to client creation (or first consensus state validation). This is somewhat equivalent to the [`Initialize()` in the spec](https://github.com/cosmos/ibc/tree/main/spec/core/ics-002-client-semantics#clientstate) and [ibc-go](https://github.com/cosmos/ibc-go/blob/61a05723b68a47a8b0249ed79088b914b0594541/modules/light-clients/07-tendermint/client_state.go#L191-L201). Should we rename to `initialize()` instead?

Reply:
Yes let's do `initialize()`

Reply:
https://github.com/informalsystems/ibc-rs/pull/2528/commits/df78cc51c8118c5bcb9c861c2f8e98ee6639123c